src/js/components/hints/symbol-hint.tsx

```diff
@@ -1,23 +1,26 @@
 'use strict';
-var React = require('react'),
-    connect = require('react-redux').connect,
-    getIn = require('../../util/immutable-utils').getIn,
-    lookup = require('../../ctrl').lookup,
-    updateMarkProperty = require('../../actions/markActions').updateMarkProperty,
-    propTypes = require('prop-types'),
-    createReactClass = require('create-react-class');
+const getIn = require('../../util/immutable-utils').getIn;
+const lookup = require('../../ctrl').lookup;
+const updateMarkProperty = require('../../actions/markActions').updateMarkProperty;
 
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {State} from '../../store';
+interface SymbolHintsProps {
+    selectedId: number,
+    updateProperty: (id: number, property: any, value: any) => void
+}
 
-function mapStateToProps(reduxState, ownProps) {
+function mapStateToProps(reduxState: State, ownProps) {
   return {
     selectedId: getIn(reduxState, 'inspector.encodings.selectedId')
   };
 }
-function mapDispatchToProps(dispatch, ownProps) {
+function mapDispatchToProps(dispatch: any, ownProps) {
```

Comment:
Dispatch is a type that you can import like
`import { Dispatch } from 'redux';`
and do `dispatch: Dispatch`

Reply:
for this one and above, don't forget `ownProps: SymbolHintsProps`